edx_lint/cmd/amnesty.py

```diff
@@ -0,0 +1,110 @@
+"""
+Commandline scripts to give one-time amnesty for specify linting error conditions.
+"""
+
+from __future__ import print_function
+
+from collections import namedtuple, defaultdict
+import re
+import sys
+
+import click
+
+PYLINT_PARSEABLE_REGEX = re.compile(
+    r"""^(?P<filename>[^:]+):(?P<line>\d+): """
+    r"""\[(?P<error_number>[^(]+)\((?P<error_code>[^)]+)\), (?P<function>[^\]]*)\] """
+    r"""(?P<error_msg>.*)"""
+)
+PYLINT_EXCEPTION_REGEX = re.compile(r"""\s*#\s*pylint:\s+disable=(?P<disables>[^#]*)""")
+
+PylintError = namedtuple('PylintError', ['filename', 'line', 'error_number', 'error_code', 'function', 'error_msg'])
+
+def parse_pylint_output(pylint_output):
+    """
+    Parse the pylint output-format=parseable lines into PylintError tuples.
+    """
+    for line in pylint_output:
+        if line.startswith("****"):
+            continue
+
+        parsed = PYLINT_PARSEABLE_REGEX.match(line)
+        if parsed is None:
+            raise Exception(
+                "Unable to parse {!r}. Please run pylint with the "
+                "--output-format=parseable option".format(
+                    line
+                )
+            )
+
+        parsed_dict = parsed.groupdict()
+        parsed_dict['line'] = int(parsed_dict['line'])
+        yield PylintError(**parsed_dict)
+
+
+def format_pylint_disables(error_codes):
+    """
+    Format a list of error_codes into a 'pylint: disable=' line.
+    """
+    if error_codes:
+        return "  # pylint: disable={}".format(", ".join(sorted(error_codes)))
+    else:
+        return ""
+
+
+def fix_pylint(line, errors):
+    """
+    Yield any modified versions of ``line`` needed to address the errors in ``errors``.
+    """
+    if not errors:
+        yield line
+        return
```

Comment:
Actually, in this case, the return is so that the function ends early (otherwise, it would just continue after the first yield).